addons/account_check_printing/models/account_payment.py

```diff
@@ -44,7 +44,7 @@ class AccountPayment(models.Model):
     )
     payment_method_line_id = fields.Many2one(index=True)
 
-    @api.constrains('check_number', 'journal_id')
+    @api.constrains('check_number', 'journal_id', 'state')
```

Comment:
As far as I've tested this is actually not working. The constrain is not called when changing the state (I bellieve this is because state field is inherited from account.move).
I believe we can remove this change
@jco-odoo

Reply:
done